experimental/GModule/src/GModule.jl

```diff
@@ -2233,8 +2388,15 @@ function split_into_homogenous(M::GModule{<:Any, <:AbstractAlgebra.FPModule{QQFi
   #         with lots of repetition
   #TODO: write and use CentreOfEndomorphismRing
   #      have a sane overall strategy
-  E, mE = endo(M)
-  C, mC = center(E)
+  #TODO: center_of_endo for sub and quo modules
+  #      irreducibles for abelian groups
+  if is_regular_gmodule(M)
+    C, mC = center_of_endo(M)
+    mE = x->x
```

Comment:
```suggestion
    mE = identity
```
using the named function `identity` instead of an anonymous function should slightly improve compilation performance